sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/_models.py

```diff
@@ -319,30 +321,32 @@ def _from_generated(cls, lifetime_action):
 class KeyRotationPolicy(object):
     """The key rotation policy that belongs to a key.
 
-    :ivar str id: The identifier of the key rotation policy.
+    :ivar id: The identifier of the key rotation policy.
+    :vartype id: Optional[str]
     :ivar lifetime_actions: Actions that will be performed by Key Vault over the lifetime of a key.
     :vartype lifetime_actions: List[~azure.keyvault.keys.KeyRotationLifetimeAction]
-    :ivar str expires_in: The expiry time of the policy that will be applied on new key versions, defined as an ISO 8601
+    :ivar expires_in: The expiry time of the policy that will be applied on new key versions, defined as an ISO 8601
         duration. For example, 90 days is "P90D".  See `Wikipedia <https://wikipedia.org/wiki/ISO_8601#Durations>`_ for
         more information on ISO 8601 durations.
+    :vartype expires_in: Optional[str]
     :ivar created_on: When the policy was created, in UTC
-    :vartype created_on: ~datetime.datetime
+    :vartype created_on: Optional[~datetime.datetime]
```

Comment:
The ivar won't be optional

Reply:
Synced about this offline -- this attribute (and others now marked as `Optional`) are actually optional, because these could be set as `None` at model creation time. `lifetime_actions` still isn't marked as `Optional` since it will always default to an empty list, even if the service returns `None` for the property